src/elements/element.arc.ts

```diff
@@ -213,13 +213,16 @@ function drawBorder(
   circular: boolean,
 ) {
   const {fullCircles, startAngle, circumference, options} = element;
-  const {borderWidth, borderJoinStyle} = options;
+  const {borderWidth, borderJoinStyle, borderDash, borderDashOffset} = options;
   const inner = options.borderAlign === 'inner';
 
   if (!borderWidth) {
     return;
   }
 
+  ctx.setLineDash(borderDash || []);
```

Comment:
You set the defat to an empty array so this check is not necessary then right?

Reply:
Yes, that was my doubt as well. But I have seen in the other parts of code, everytime this check is done for borderDash because setLineDash fails if not an array, I guess. Therefore not clear to me if it's not correct on the other parts or if better to leave this check anyway :(

Reply:
@LeeLenaleee don't ask me why because I don't have the answer now (I checked this morning without success) but it seems that the donut options doesn't fallback to element ones therefore if you set `borderDash: null` in dataset config (and you don't have those check), you have an issue. This weird behavior was already recognized I guess because the `_indexable` about `spacing` option was added to the controller instead of the element. If you agree, I can pend more time to go in deep why the donut doesn't fallback to the arc element (maybe in another PR)

Reply:
@LeeLenaleee I have checked more in deep and the descriptors are working well. I wasn't aware that the resolver uses the descriptors of the default of the controller (and not in cascade as I was thinking) therefore it's working as designed.

About the point of your review, I can confirm that if you set `borderDash: null`, you have an error (but fallback works correctly if you set to `undefined`).
The check where null is passed as defined is here:

https://github.com/chartjs/Chart.js/blob/e417c60c22c1d38fcf5f2b1a9340a4a90ff946c4/src/helpers/helpers.config.js#L331

The `define` function checks if the value is `undefined` but with `null` argument returns `true`. Probably the `define` function could be remove, using instead `isNullOrUndef`. WDYT?

In my opinion, but I can be wrong, it would be better to maintain that check anyway in the meanwhile. But feel free to disagree ;)

Reply:
Null is considered a valid value for configuration while undefined is considered not defined and thus fallback. Falling back from null would be a breaking change, extending to all unknown use cases.

Reply:
@kurkle yes!

Reply:
I kind of agree with @LeeLenaleee, but the behaviour should be changed everywhere to be consistent, so maybe do at v5 as a breaking change.

Reply:
@kurkle @LeeLenaleee @etimberg I think we agreed to change for v5. Nevertheless let me outline that there are other options/situation which should be addressed, similar to this one, I guess. Maybe you are already aware but let me make an example.

The doughnut controller is setting `spacing` options in the default, set to 0. doing that, the `spacing` option of the element is ignored, therefore having a following chart config

```javascript
{
  type: 'doughnut',
  data: {
    labels: [0, 1, 2, 3, 4, 5],
    datasets: [
      {
        data: [0, 2, 4, null, 6, 8],
        cutout: 200
      }]
  },
  options: {
    elements: {
      arc: {
        spacing: 30
      }
    }
  }
}
```

the `spacing: 30` is ignored in the cutout/maxsize calculation because it uses 0 (controller defaults) instead of 30. This is happening because the dataset options are not using the fallback in the update phase of the controller and when you have to use options of data element during the update of the controller, you must check against a default or replicate the options definition, ignoring the element config value/defaults. @kurkle I think we have already talked about that in the treemap controller.